crates/sui/src/sui_commands.rs

```diff
@@ -42,17 +50,121 @@ use sui_swarm_config::network_config_builder::ConfigBuilder;
 use sui_swarm_config::node_config_builder::FullnodeConfigBuilder;
 use sui_types::base_types::SuiAddress;
 use sui_types::crypto::{SignatureScheme, SuiKeyPair, ToFromBytes};
+use tempfile::tempdir;
+use tracing;
 use tracing::info;
 
+const CONCURRENCY_LIMIT: usize = 30;
+const DEFAULT_EPOCH_DURATION_MS: u64 = 60_000;
+const DEFAULT_FAUCET_NUM_COINS: usize = 5; // 5 coins per request was the default in sui-test-validator
+const DEFAULT_FAUCET_MIST_AMOUNT: u64 = 200_000_000_000; // 200 SUI
+
 #[allow(clippy::large_enum_variant)]
 #[derive(Parser)]
 #[clap(rename_all = "kebab-case")]
 pub enum SuiCommand {
-    /// Start sui network.
+    /// Start a local network in two modes: saving state between re-runs and not saving state
+    /// between re-runs. Please use (--help) to see the full description.
+    ///
+    /// By default, sui start will start a local network from the genesis blob that exists in
+    /// the Sui config default dir or in the config_dir that was passed. If the default directory
+    /// does not exist and the config_dir is not passed, it will generate a new default directory,
+    /// generate the genesis blob, and start the network.
+    ///
+    /// Note that if you want to start an indexer, Postgres DB is required.
     #[clap(name = "start")]
     Start {
+        /// Config directory that will be used to store network config, node db, keystore
+        /// sui genesis -f --with-faucet generates a genesis config that can be used to start this
+        /// proces. Use with caution as the `-f` flag will overwrite the existing config directory.
+        /// We can use any config dir that is generated by the `sui genesis`.
         #[clap(long = "network.config")]
-        config: Option<PathBuf>,
+        config_dir: Option<std::path::PathBuf>,
+
+        /// A new genesis is created each time this flag is set, and state is not persisted between
+        /// runs. Only use this flag when you want to start the network from scratch every time you
+        /// run this command.
+        ///
+        /// To run with persisted state, do not pass this flag and use the `sui genesis` command
+        /// to generate a genesis that can be used to start the network with.
+        #[clap(long)]
+        force_regenesis: bool,
+
+        /// Start a GraphQL server on localhost and port: 127.0.0.1:9125, or on the port provided.
+        /// When providing a specific value, please use the = sign between the flag and value:
+        /// `--with-graphql=6125`.
+        /// Note that GraphQL requires a running indexer.
```

Comment:
Can we make `--with-indexer` implicit, if `--with-graphql` is provided?

Reply:
We can but I usually prefer explicit vs implicit in runtime variability so that a user knows exactly what it's running.

Reply:
Fixed

Reply:
Yeah up to you, I can see arguments either way. 

For me the motivation for this suggestion was the fact that the tool tells you what it's starting up, which helps counter the confusion from the implicit behaviour, and making it explicit only offers someone the possibility to make a mistake.